nodejs/week2/meal-sharing/src/backend/api/meals-router.js

```diff
@@ -0,0 +1,68 @@
+const { query } = require("express");
+const express = require("express");
+const router = express.Router();
+
+const meals = require("./../data/meals.json");
+
+router.get("/", async (request, response) => {
+  try {
+    const supportedParams = ["maxPrice", "limit", "title", "createdAfter"]; //will be needed for query validation
+    const maxPrice = parseInt(request.query.maxPrice); //converting to int
+    const limit = parseInt(request.query.limit); //converting to int
+    const title = request.query.title;
+    const createdAfter = request.query.createdAfter;
+
+    //Checking if there are any queries
+    if (Object.keys(request.query) != 0) {
+      if (maxPrice) {
+        response.send(meals.filter((meal) => meal.price <= maxPrice));
+      } else if (title) {
+        response.send(
+          meals.filter(
+            (meal) => meal.title.toLowerCase().includes(title.toLowerCase()) //getting rid of case sensitivity
+          )
+        );
+      } else if (createdAfter) {
+        if (new Date(createdAfter) != "Invalid Date") {
+          response.send(
+            meals.filter(
+              (meal) => new Date(meal.createdAt) > new Date(createdAfter)
+            )
+          );
+        } else {
+          response.send(400);
+        }
+      } else if (limit) {
+        response.send(meals.slice(0, limit));
+      } else if (!supportedParams.includes(request.query)) {
+        response.send(400, "This query parameter is not supported");
+      }
+    } else {
+      response.send(meals);
+    }
+  } catch (error) {
+    throw error;
```

Comment:
What would happen if there are more than one valid parameter in the request? Can you modify your implementation to take all of them into account?

Reply:
fixed it :)